woocommerce-gateway-ebanx/includes/ebanx-php/src/Ebanx/Http/ClientCurl.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright (c) 2017, EBANX Tecnologia da Informação Ltda.
+ *  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * Neither the name of EBANX nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Ebanx\Http;
+
+/**
+ * HTTP request client wrapper, using curl
+ *
+ * @author Guilherme Pressutto guilherme.pressutto@ebanx.com
+ */
+class ClientCurl extends AbstractClient
+{
+    private $curl;
+
+    /**
+     * {@inheritDoc} using curl.
+     */
+    public function send()
+    {
+        $this->_setupCurl();
+        $response = curl_exec($this->curl);
+        if(curl_getinfo($this->curl, CURLINFO_HTTP_CODE) === 200) {
+            if($this->hasToDecodeResponse) {
+                $response = json_decode($response);
+            }
+        } else {
+            if(curl_errno($this->curl)) {
```

Comment:
There are a few weird responses that don't use 200, in case of api business rules validations.
The ignoredStatusCodes array was built to filter them out.